lib/dangermattic/plugins/labels_pr_checker.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Danger
+  # Plugin to check for labels in a PR.
+  class LabelsPRChecker < Plugin
+    # Checks if a PR is missing labels or is marked as 'do not merge'.
+    def check_labels
+      github_labels = danger.github.pr_labels
+
+      warn('PR is missing at least one label.') if github.pr_labels.empty?
```

Comment:
Might be for a subsequent PR—when you'll consider refactoring some existing rules as opposed to just porting the existing Peril ones _as is_ like here—but it could be worth merging this Danger rule with the future port of https://github.com/Automattic/peril-settings/blob/master/org/issue/label-woo.ts and https://github.com/Automattic/peril-settings/blob/master/org/issue/label.ts to make a single, more flexible/generic rule instead.

For example, in WCiOS, we have a set of GitHub Labels prefixed by `feature:`, and also a set of labels prefixed with `type:`, and we require all issues to have at least one of those `feature: *` labels and one of those `type: *` labels.

Maybe your dangermattic Plugin here could factor all that into one single plugin that could be applied to both PRs and Issues, and configurable for each repo's labels, instead of being specific for PRs with the hardcoded value of Do Not Merge?

I'm thinking something like this for the call site in the client's Dangerfile:

```ruby
labels_checker.do_not_merge_label = 'Do Not Merge'
labels_checker.require_label(/feature: .*/)
labels_checker.require_label(/type: .*/)
labels_checker.check()
```

Or something like:
```ruby
labels_checker.check(
  do_not_merge_label: 'Do Not Merge',
  required_labels: [/feature: .*/, /type: .*/]
)
```

Reply:
```
your dangermattic Plugin here could factor all that into one single plugin that could be applied to both PRs and Issues, and configurable for each repo's labels, instead of being specific for PRs with the hardcoded value of Do Not Merge
```

That's exactly what I think as well -- some of the previous Peril rules are perhaps too tied to a specific process or even specific app practices. We should try to generalise them to make them applicable at least to all A8c apps.

For now this was a 1:1 port, but I think your suggestion makes a lot of sense and shouldn't be a lot of trouble to generalise from the beginning in this case. Let me give it a try!

Reply:
I've changed the plugin to generalise it a bit more and receive these parameters.